pkg/testutil/kopia_testutil.go

```diff
@@ -0,0 +1,216 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package testutil
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"math/big"
+	"os"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	crv1alpha1 "github.com/kanisterio/kanister/pkg/apis/cr/v1alpha1"
+	awsconfig "github.com/kanisterio/kanister/pkg/aws"
+)
+
+const (
+	DefaultRepositoryPath                 = "kopia-repo-controller-test"
+	DefaultRepositoryServerAdminUser      = "admin@test"
+	DefaultRepositoryServerHost           = "localhost"
+	DefaultRepositoryPassword             = "test1234"
+	DefaultKanisterAdminUser              = "kanisterAdmin"
+	DefaultKanisterUser                   = "kanisteruser"
+	DefaultKanisterNamespace              = "kanister"
+	DefaultKopiaRepositoryServerContainer = "repo-server-container"
+	ServerReady                           = "ServerReady"
+)
+
+func S3CredsLocationSecret() (*v1.Secret, *v1.Secret) {
+	key := os.Getenv(awsconfig.AccessKeyID)
+	val := os.Getenv(awsconfig.SecretAccessKey)
+	s3Creds := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "test-s3-creds-",
+			Labels: map[string]string{
+				"repo.kanister.io/target-namespace": "monitoring",
+			},
+		},
+		Type: "secrets.kanister.io/aws",
+		Data: map[string][]byte{
+			"aws_access_key_id":     []byte(key),
+			"aws_secret_access_key": []byte(val),
+		},
+	}
+	s3Location := &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "test-s3-location-",
+			Labels: map[string]string{
+				"repo.kanister.io/target-namespace": "monitoring",
+			},
+		},
+		Data: map[string][]byte{
+			"type":     []byte(crv1alpha1.LocationTypeS3Compliant),
+			"bucket":   []byte(TestS3BucketName),
+			"path":     []byte(DefaultRepositoryPath),
+			"region":   []byte(TestS3Region),
+			"endpoint": []byte(os.Getenv("LOCATION_ENDPOINT")),
+		},
+	}
+	return s3Creds, s3Location
+}
+
+func KopiaRepositoryPassword() *v1.Secret {
+	return &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "test-repo-pass-",
+		},
+		Data: map[string][]byte{
+			"repo-password": []byte(DefaultRepositoryPassword),
+		},
+	}
+}
+
+func KopiaRepositoryServerAdminUser() *v1.Secret {
+	return &v1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "test-repository-admin-user-",
+		},
+		Data: map[string][]byte{
+			"username": []byte(DefaultRepositoryServerAdminUser),
+			"password": []byte(DefaultRepositoryPassword),
```

Comment:
The password here should be different from what repository password is. `DefaultRepositoryPassword` is used to connect to repository whereas the password here required is kopia repository server admin password

Reply:
Yeah, Created new constant